simvue/filter.py

```diff
@@ -0,0 +1,231 @@
+
+    def as_list(self) -> list[str]:
+        return self._filters
+
+    def clear(self) -> None:
+        self._filters = []
+
+
+class FoldersFilter(RestAPIFilter):
+    def has_path(self, name: str) -> "FoldersFilter":
+        self._filters.append(f"path == {name}")
+        return self
+    
+    def has_path_containing(self, name: str) -> "FoldersFilter":
+        self._filters.append(f"path contains {name}")
+        return self
+    
+    def _generate_members(self) -> None:
+        return super()._generate_members()
+
+
+class RunsFilter(RestAPIFilter):
+    def _generate_members(self) -> None:
+        _global_comparators = [
+            self._value_contains,
+            self._value_eq,
+            self._value_neq
+        ]
+
+        _numeric_comparators = [
+            self._value_geq,
+            self._value_leq,
+            self._value_lt,
+            self._value_gt
+        ]
+
+        for label, system_spec in System.__members__.items():
+            for function in _global_comparators:
+                _label: str = label.lower()
+                _func_name: str = function.__name__.replace("_value", _label)
+                _out_func = lambda value, func=function: func("system", system_spec.value, value)
+                _out_func.__name__ = _func_name
+                setattr(self, _func_name, _out_func)
+
+        for function in _global_comparators + _numeric_comparators:
+            _func_name: str = function.__name__.replace("_value", "metadata")
+            _out_func = lambda attribute, value, func=function: func("metadata", attribute, value)
+            _out_func.__name__ = _func_name
+            setattr(self, _func_name, _out_func)
+
+    def author(self, username: str="self") -> "RunsFilter":
+        self._filters.append(f"user == {username}")
+        return self
+
+    def exclude_author(self, username: str="self") -> "RunsFilter":
+        self._filters.append(f"user != {username}")
+        return self
+    
+    def starred(self) -> "RunsFilter":
+        self._filters.append("starred")
+        return self
+    
+    def has_name(self, name: str) -> "RunsFilter":
+        self._filters.append(f"name == {name}")
+        return self
+    
+    def has_name_containing(self, name: str) -> "RunsFilter":
+        self._filters.append(f"name contains {name}")
+        return self
+
+    def has_status(self, status: Status) -> "RunsFilter":
+        self._filters.append(f"status == {status.value}")
+        return self
+    
+    def is_running(self) -> "RunsFilter":
+        return self.has_status(Status.Running)
+    
+    def is_lost(self) -> "RunsFilter":
+        return self.has_status(Status.Lost)
+    
+    def has_completed(self) -> "RunsFilter":
+        return self.has_status(Status.Completed)
+    
+    def has_failed(self) -> "RunsFilter":
+        return self.has_status(Status.Failed)
+    
+    def has_alert(self, alert_name: str, is_critical: typing.Optional[bool]=None) -> "RunsFilter":
```

Comment:
I don't think this is the correct format for a filter that requires an alert exists and is in a specific state. I think those two subparts should be a single filter (connected with `&&`). See: https://github.com/simvue-io/server/blob/f02a2866c64185736955763725b26b108520f540/simvue/core/run.py#L618 

I do think that it could be worth updating the server to use the format used here (or an different 3rd format) instead.